src/RequireJS/BundleGenerator/ConfigGen.tsx

```diff
@@ -6,17 +6,74 @@
 import * as React from 'react';
 import { RequireConfig, PageModule } from '../../types/require';
 import generate from './generate';
+import JavaScriptHighlighter from '../../JavaScriptHighlighter';
+import styles from './ConfigGen.css';
 
 type Props = {
     pageModules: PageModule[];
     requireConfig: RequireConfig;
 };
 
-export default class ConfigGen extends React.Component<Props> {
-    render() {
+export default class ConfigGen extends React.PureComponent<Props> {
+    generateJS() {
         const { pageModules, requireConfig } = this.props;
         const config = generate(pageModules, requireConfig);
+        const code = JSON.stringify(config, null, 4);
+        return transforms.reduce((code: string, transform) => {
+            return transform(code);
+        }, code);
+    }
 
-        return <pre>{JSON.stringify(config, null, 2)}</pre>;
+    render() {
+        const code = this.generateJS();
+        return (
+            <JavaScriptHighlighter code={code} className={styles.codeView} />
+        );
     }
 }
+
+const transforms = [
+    parenWrap,
+    addConfigReference,
+    sharedBundleComment,
+    uglifyComment,
+];
+
+function parenWrap(code: string) {
+    return `(${code})`;
+}
+
+function addConfigReference(code: string) {
+    return `// Full config reference:
+// https://github.com/requirejs/r.js/blob/b8a6982/build/example.build.js
+
+${code}`;
+}
+
+function sharedBundleComment(code: string) {
+    const byLine = code.split('\n');
+    const sharedBundleIndex = byLine.findIndex(c =>
+        c.includes('"name": "bundles/shared",'),
+    );
+    const [whitespace] = byLine[sharedBundleIndex].match(/(\s+)/) || [''];
+    byLine.splice(
+        sharedBundleIndex,
+        0,
+        `${whitespace}// Modules used on > 1 page(s) of the store`,
+    );
+    return byLine.join('\n');
+}
+
+function uglifyComment(code: string) {
```

Comment:
This function shares a lot of the same logic as `sharedBundleComment`. Recommend consolidation.

Reply:
Big big big proponent of the [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming))

Reply:
⚾️